src/main/kotlin/org/malachite/estella/offer/api/OfferController.kt

```diff
@@ -19,67 +21,58 @@ import java.util.*
+        offerService.getOffers()
+            .map { it.toOfferResponse() }
+            .let { ResponseEntity(it, HttpStatus.OK) }
+
 
     @CrossOrigin
     @GetMapping("/{offerId}")
-    fun getOffer(@PathVariable offerId: Int): ResponseEntity<OfferResponse> {
-        val offer = offerService.getOffer(offerId)
-
-        return ResponseEntity(OfferResponse.fromOffer(offer), HttpStatus.OK)
-    }
+    fun getOffer(@PathVariable offerId: Int): ResponseEntity<OfferResponse> =
+        offerService.getOffer(offerId)
+            .let { ResponseEntity(it.toOfferResponse(), HttpStatus.OK) }
 
     @CrossOrigin
     @PostMapping("/addoffer")
-    fun addOffer(@RequestBody offer: OfferRequest, @CookieValue("jwt") jwt: String?): ResponseEntity<Any> {
-        val hrPartner =
-            securityService.getHrPartnerFromJWT(jwt) ?: return ResponseEntity.status(404).body("Unauthenticated")
-
-        val saved: Offer = offerService.addOffer(offer.toOffer(hrPartner, desiredSkillService))
-
-        val recruitmentProcess = RecruitmentProcess(
-            null,
-            Date.valueOf(LocalDate.now()),
-            null,
-            saved,
-            listOf(RecruitmentStage(null, StageType.APPLIED)),
-            setOf(), setOf()
-        )
-        recruitmentProcessService.addProcess(recruitmentProcess)
-
-        return ResponseEntity.created(URI("/api/offers/" + saved.id)).build()
+    fun addOffer(
+        @RequestHeader(EStellaHeaders.jwtToken) jwt: String?,
+        @RequestBody offer: OfferRequest
+    ): ResponseEntity<Message> {
+        val hrPartner = getHrPartnerFromJWT(jwt)
+        offerService.addOffer(offer, hrPartner)
+        return ResponseEntity(SuccessMessage, HttpStatus.OK)
     }
 
     @CrossOrigin
     @PutMapping("/update/{offerId}")
-    fun updateOffer(@PathVariable("offerId") offerId: Int, @RequestBody offer: OfferRequest,
-                    @CookieValue("jwt") jwt: String?): ResponseEntity<Any> {
-        val hrPartner =
-            securityService.getHrPartnerFromJWT(jwt) ?: return ResponseEntity.status(404).body("Unauthenticated")
-        offerService.updateOffer(offerId, offer.toOffer(hrPartner, desiredSkillService))
-        return ResponseEntity(HttpStatus.OK)
+    fun updateOffer(
+        @RequestHeader(EStellaHeaders.jwtToken) jwt: String?,
+        @PathVariable("offerId") offerId: Int,
+        @RequestBody offer: OfferRequest
+    ): ResponseEntity<Any> {
+        val hrPartner = getHrPartnerFromJWT(jwt)
+        offerService.updateOffer(offerId, offer, hrPartner)
+        return ResponseEntity(SuccessMessage, HttpStatus.OK)
     }
 
     @CrossOrigin
     @DeleteMapping("/{offerId}")
-    fun deleteOffer(@PathVariable("offerId") offerId: Int): ResponseEntity<Offer> {
+    fun deleteOffer(
+        @RequestHeader(EStellaHeaders.jwtToken) jwt: String?,
+        @PathVariable("offerId") offerId: Int
+    ): ResponseEntity<Message> {
+        val hrPartner = getHrPartnerFromJWT(jwt)
```

Comment:
Nie używamy nigdzie tego hrPartnera obecnie, więc pytanie czy chwilowo pomijamy JWT? Bo w testach przekazujemy token.

Reply:
Ogólnie sama definicja getHrPartner walnie unautha, jeśli jest on nullowy - zmiennej faktycznie na razie nie używamy, to ją wywalę.